### ISSUE TITLE
preview-auto is fantastic

### ISSUE THREAD

**Issue by aloispichler** [ultronozm/preview-auto.el#1]
Just wanted to let you know that your preview functionality is certainly the best I have encountered so far. 
This here comes with equation numbers, which is handy as well ;-) https://www.youtube.com/watch?v=n-AfvuV-bYo
Thanks!

**Comment by ultronozm** [comment]
Thanks Alois, appreciate the kind words!

Regarding equation numbering, you just need to set up the package
auctex-label-numbers (available via ELPA) and keep the aux file up to
date (e.g., via the package auctex-cont-latexmk with which you're
already familiar).  This is a bit different from what I think you get
with org-latex-preview, where the numbering is consistent in any buffer
but not (afaik) faithful to the numbering in a compiled tex document.
Let me know if you have any difficulties.

Paul

**Comment by aloispichler** [comment]
Hello again!
I saw your twitch today and realized that it is not running here as on your screen:
Adding a new `$…$` or `\[ … \]` will add the preview, this is fine.
But returning to the formula and changing inside `$…$` does not auto-update. Hm, but it seems this is the main functionality… I could not get this running here.

**Comment by aloispichler** [comment]
Hello again.
I was playing a bit now:
* With `(setq preview-leave-open-previews-visible nil)`, the previews are correctly updated (but not visible during the edits).
* With `(setq preview-leave-open-previews-visible t)`, the previews fall back to the initial version, they do not show the new edits.

I hope this observation is helpful. Thanks!

**Comment by ultronozm** [comment]
Thanks Alois, I'm thinking about this.

Out of curiosity, do you edit tex files in a shared folder (e.g., Dropbox)?

You might also try checking if the issue really concerns AUCTeX rather than
preview-auto, e.g., by setting the option that you just mentioned to t, and
then recompiling them in place via `C-c C-p C-e`, for instance.

**Comment by aloispichler** [comment]
Yes, that’s on Dropbox. Hmm, does it matter?

**Comment by ultronozm** [comment]
I've seen some odd things happen with Emacs on shared drives.  I think it
still counts as a bug, but might be worth checking without Dropbox, just to
help diagnose the issue

**Comment by aloispichler** [comment]
Indeed, I just tried it and it seems to work outside of Dropbox… Good catch!

**Comment by ultronozm** [comment]
Copy.  Maybe try the test with just AUCTeX (and no preview-auto) that I
mentioned; let me know if it's confusing

**Comment by aloispichler** [comment]
In Dropbox, `(setq preview-leave-open-previews-visible t)` and `(preview-auto)`: 
_without_ recompiling: does not change the display, wrong preview.

In Dropbox, `(setq preview-leave-open-previews-visible t)` and `(preview-auto)`: 
_with_ recompiling: does not change the display, as above.

In Dropbox, `(setq preview-leave-open-previews-visible t)` and `(preview-auto)` disabled. Correct display after `M-x preview-buffer`.

Is this the desired test?

**Comment by ultronozm** [comment]
For the cases where you test with `preview-auto`, what do you mean by
"recompiling"?

For the test without `preview-auto`, what happens if you do
`preview-environment` instead of `preview-buffer`?

**Comment by aloispichler** [comment]
* Recompile= save the file, which triggers (your auctex-cont-)latexmk to compile.
* In Dropbox, `(setq preview-leave-open-previews-visible t)` and `(preview-auto)` disabled:
I have encountered now wrong displays after `M-x preview-buffer` as well.
But the displays after `M-x preview-environment` generally do not update, they are wrong almost always.
* I haven’t seen problems in nextcloud, by the way, just with dropbox.

Thanks. Hope this is helpful.

**Comment by ultronozm** [comment]
OK, very good.  So if I've followed correctly, it's an issue with the
preview-leave-open-previews-visible option in AUCTeX (which I'm responsible
for having added) together with Dropbox.  Let me know if I've misunderstood.

It might also be helpful if you could share your system info.

I don't have the issue on my computer, so I'll have to think about what
might be causing it.

**Comment by aloispichler** [comment]
I am on Arch, hyprland, Emacs 29.4.

**Comment by aloispichler** [comment]
Here is a question where I would appreciate your help: I have been using `TeX-view` (C-c C-v) alot. However, this does not work with preview. I am getting the message `…/_region_.pdf`  does not exist. Any ideas how to get `TeX-view` running with preview??? Many thanks!

**Comment by ultronozm** [comment]
Hi Alois, for your last question, the docs for `TeX-view` say that it operates on the region or master file, depending upon the last command issued.  So I guess that to use it, you'd want to disable preview-auto and compile the document using C-c C-c.  But depending upon what you want, something more efficient could be hacked together.

Also, maybe this discussion would be better continued on one of the AUCTeX mailing lists?  I'm a regular there, so we can continue it just the two of us, but maybe the others would chime in with other ideas.

**Comment by ultronozm** [comment]
Hi Alois,

For the problem you mentioned earlier, about `(setq preview-leave-open-previews-visible t)` and Dropbox, could you try plopping the following in your config, and seeing if it does anything about the issue?

```elisp
(with-eval-after-load 'preview
  (defun preview-replace-active-icon (ov replacement)
    "Replace the active Icon in OV by REPLACEMENT, another icon."
    (let ((disp (overlay-get ov 'preview-image)))
      (when (consp disp) (image-flush disp t)))
    (overlay-put ov 'preview-image (copy-tree replacement))))
```

**Comment by aloispichler** [comment]
I have tested this very briefly only, but – magically – this seems to work :-)
Great!

**Comment by ultronozm** [comment]
Great!  Let me know if any issues come up -- I've had some weird errors with this, but glad that it seems to work for now.

**Comment by aloispichler** [comment]
Okay. Let me know once this is fixed in AUCTeX or your beautiful extension.
